libsolutil/CommonData.cpp

```diff
@@ -151,11 +151,11 @@ string solidity::util::getChecksummedAddress(string const& _addr)
 	for (size_t i = 0; i < 40; ++i)
 	{
 		char addressCharacter = s[i];
-		unsigned nibble = (unsigned(hash[i / 2]) >> (4 * (1 - (i % 2)))) & 0xf;
+		size_t nibble = static_cast<size_t>(hash[static_cast<unsigned>(i) / 2] >> (4 * (1 - (i % 2)))) & 0xf;
```

Comment:
Why `size_t` if you're casting to `unsigned` inside the parenthesis?

Reply:
operator `[]` for `hash` needs unsigned as parameter. Maybe changing `i` to unsigned will solve it nicer